package/network/services/hostapd/files/hostapd.sh

```diff
@@ -408,7 +409,7 @@ hostapd_set_psk_file() {
 	json_get_vars mac vid key
 	set_default mac "00:00:00:00:00:00"
 	[ -n "$vid" ] && vlan_id="vlanid=$vid "
-	echo "${vlan_id} ${mac} ${key}" >> /var/run/hostapd-${ifname}.psk
+	echo "${vlan_id}${mac} ${key}" >> /var/run/hostapd-${ifname}.psk
```

Comment:
This is wrong, there needs to be a space between vlanid and macaddr. This won't work.

Reply:
A space character is already set at the previous line: if `vlanid` is set `vlan_id` will be inserted followed by a space, if not a space will not be inserted before `mac`.

Reply:
I missed that, you're right. Maybe it's personal preference but I think it's nicer to remove the space from $vlan_id and keep the space in the echo command. I find it clearer that way.

Reply:
That was my idea first, I find it clearer as well.
While not a problem per se it will insert a leading space before `mac` when `vid` is empty... Alternatively, it could be modified as follows:

```bash
set_default mac "00:00:00:00:00:00"
local psk_file_entry="${mac} ${key}"
[ -n "$vid" ] && psk_file_entry="vlanid=${vid} ${psk_file_entry}"
echo "${psk_file_entry}" >> /var/run/hostapd-${ifname}.psk
```

Reply:
I'd just remove this change to keep the scope towards sae_password_file, though I personally think it would be much better to just backport the patch that added sae_password_file in hostapd instead of this.

Reply:
P.S. if you don't know how, you could cherry-pick this commit: https://github.com/rany2/openwrt-1/commit/3788bf8e602c264db18737de9e65521d37bc0aed

Reply:
Just throwing it in:
```
hostapd_set_psk_file() {
        ..
        [ -n "$vid" ] && echo -n "vlanid=${vid} "
        echo "${mac} ${key}"
}

hostapd_set_psk() {
	local ifname="$1"
	
	for_each_station hostapd_set_psk_file "${ifname}" > /var/run/hostapd-${ifname}.psk
}
```

Free for the taking :wink: